src/resmom/start_exec.c

```diff
@@ -3577,7 +3571,7 @@ start_process(task *ptask, char **argv, char **envp, bool nodemux)
 		case 0:	/* explicit reject */
 			free(progname);
 			free_attrlist(&argv_list);
-			free(env_str);
+			free_str_array(hook_output.env);
 			starter_return(kid_write, kid_read,
 					JOB_EXEC_FAILHOOK_DELETE, &sjr);
```

Comment:
starter_return() will do an exit(254) if the third argument (code) is < 0. In this case, JOB_EXEC_FAILHOOK_DELETE is -17 so this will cause an exit(). This is being executed in a child of main mom acting as the job.